Aliases/Server - Heroism & Shame/dt/animal.alias

```diff
@@ -0,0 +1,101 @@
+embed -thumb <image> -color <color> -footer "Usage - !dt animal -name \"name\" [adv] [-rr #]"
+<drac2>
+# sets primary variables
+me, pargs, returnstr, trainComplete, trainCounter = character(), argparse(&ARGS&), "", False, 0
+
+# sets the booleans
+withAdv = True if pargs.get("adv") else False
+hasAnimal = True if exists("DTAnimalTrain") else False
+choseAnimal = True if pargs.get("name") else False
+
+# sets roll string
+rollStr = "1d20+" if not withAdv else "2d20kh1+"
+rollStr += str(me.skills.animalHandling.value)
+# sets the repeats
+repeats = int(pargs.get("rr")[0]) if pargs.get("rr") else 1
+
+# sets the error strings
+errorstr = f' -title "{me.name} attempts to do Animal Training"' \
+           f' -f "ERROR | Animal not set up, or you didn\'t use `-name \"name\"`. Go do `!dt animal setup` or spell it right"' \
+
+# does the error checking
+if not hasAnimal:
+    return errorstr
+elif not choseAnimal:
+    return errorstr
+else:
+    chosenAnimal = pargs.get("name")[0]
+
+#load the cvar
+animalList = load_json(DTAnimalTrain)
+
+#checks to see if the passed animal is there
+if chosenAnimal not in animalList.keys():
+    for aname, ainfo in animalList.items():
+        if chosenAnimal in ainfo.values():
+            myAnimal = animalList[aname]
+            chosenAnimal = aname
+            break
+else:
+    myAnimal = animalList[chosenAnimal]
+if not exists("myAnimal"):
+    return errorstr
+ccName, currentLevel, dc= f'Animal Training-{chosenAnimal}', myAnimal["traininglevel"], myAnimal["dc"]
+nextLevel = currentLevel
+returnstr += f' -title "{me.name} trains {myAnimal["name"]}" -desc "Current trained level: `{currentLevel}`" '
+#errors checks to see if there is no more training to be done
+if myAnimal["traininglevel"] == "drilled":
+    returnstr += f' -f "COMPLETE| **ALL** Training Complete, No more need to train {myAnimal["name"]}."'
+    return returnstr
+# runs the training portion
+for x in range(repeats):
+    # first checks to see if the CC has been completed, otherwise, wont roll
+    if me.cc(ccName).value >= me.cc(ccName).max or trainComplete:
+        returnstr += f' -f "COMPLETE| Training Complete, will not roll anymore."'
+        trainComplete = True
+        break
+    #builds the roll and checks to see if it passed
+    animalroll = vroll(rollStr)
+    crit = animalroll.result.crit
+    critstr = "Fail"
+    passedRoll = animalroll.total >= dc
+    if passedRoll and not trainComplete:
+        if not crit == -1:
```

Comment:
There is no crit_type with -1

Reply:
bruh i copied your example from embodiment. it does work it trues on 0